jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/SchemaStoreTest.java

```diff
@@ -76,6 +76,20 @@ public void createWithSelfRef() throws URISyntaxException {
     }
 
     @Test
+    public void createWithEmbeddedSelfRef() throws URISyntaxException {
+
+        URI schemaUri = getClass().getResource("/schema/embeddedRef.json").toURI();
+
+        SchemaStore schemaStore = new SchemaStore();
+        Schema topSchema = schemaStore.create(schemaUri);
+        Schema embeddedSchema = schemaStore.create(topSchema, "#/definitions/embedded");
+        Schema selfRefSchema = schemaStore.create(embeddedSchema, "#");
+
+        assertThat(topSchema, is(sameInstance(selfRefSchema)));
```

Comment:
@joelittlejohn This is the core thing that I think I have fixed here. Do you agree with this test?